tests/resources/test_project.py

```diff
@@ -45,15 +57,40 @@ def project(session) -> Project:
     return project
 
 
+@pytest.fixture
+def project_v3(session_v3) -> Project:
+    project = Project(
+        name='Test Project',
+        session=session_v3,
+        team_id=uuid.UUID('11111111-8baf-433b-82eb-8c7fada847da')
+    )
+    project.uid = uuid.UUID('16fd2706-8baf-433b-82eb-8c7fada847da')
+    return project
+
+
 @pytest.fixture
 def collection(session) -> ProjectCollection:
     return ProjectCollection(session)
 
 
+@pytest.fixture
+def collection_v3(session_v3,) -> ProjectCollection:
+    return ProjectCollection(session_v3, team_id=uuid.uuid4())
+
+
 def test_string_representation(project):
     assert "<Project 'Test Project'>" == str(project)
 
 
+def test_share_post_content_v3(project_v3):
+    # Given
+    dataset_id = str(uuid.uuid4())
+    other_team_id = uuid.uuid4()
```

Comment:
Why is one of these a str and the other a UUID?

Reply:
the share method takes `project_id: Optional[Union[str, UUID]]` but only `resource_id: Optional[str]` 🤷‍♀️ (it's an existing method, so I didn't change the signature) 
In particular, I think UUIDs aren't json serializable. 

But that should be other _project_id, instead of other_team_id